sympy/core/function.py

```diff
@@ -264,7 +264,7 @@ def __new__(cls, *args, **options):
         options.pop('nargs', None)
 
         if options:
-            raise ValueError("Unknown options: %s" % options)
+            return cls.eval(*args, **options)
 
         if evaluate:
             evaluated = cls.eval(*args)
```

Comment:
`**options` should be added here. The `if options` block above should just be removed. This change doesn't respect the `evaluate` flag if there are other options (`log(4, base=2, evaluate=False)`).

Reply:
In that case, if evaluate is `false` then it will be raising same error.

Reply:
It should give the same thing as `log(4, 2, evaluate=False)`. 

Apparently this currently returns a 2-argument log, which is problematic since it isn't actually supported (see https://github.com/sympy/sympy/issues/6888). But even so, that is unrelated to this pull request.

Reply:
@RituRajSingh878 I see, if evaluate=False, it won't use the signature of `eval`, so it cannot know that the keyword arguments are actually positional. I think we could make it work, perhaps by pulling the signature of `eval`. Although it would also be fine to have

```py
if not evaluate and options:
    raise NotImplementedError("keyword arguments with evaluate=False are not supported")
```

Reply:
@asmeurer this can be a better option.


> 
> ```python
> if not evaluate and options:
>     raise NotImplementedError("keyword arguments with evaluate=False are not supported")
> ```